bin/copySourceMemberContentPreference.ts

```diff
@@ -0,0 +1,44 @@
+import '../src/config';
+import createOrGetConnection from '../src/db';
+
+(async (): Promise<void> => {
+  const limitArgument = process.argv[2];
+  const offsetArgument = process.argv[3];
+
+  if (!limitArgument || !offsetArgument) {
+    throw new Error('limit and offset arguments are required');
+  }
+
+  const limit = +limitArgument;
+
+  if (Number.isNaN(limit)) {
+    throw new Error('limit argument is invalid, it should be a number');
+  }
+
+  const offset = +offsetArgument;
+
+  if (Number.isNaN(offset)) {
+    throw new Error('offset argument is invalid, it should be a number');
+  }
+
+  const con = await createOrGetConnection();
+
+  await con.transaction(async (manager) => {
+    await manager.query(`SET session_replication_role = replica;`);
+
+    await manager.query(`
+      INSERT INTO content_preference ("userId", "referenceId", "type", "createdAt", "status", "sourceId", "flags")
+      SELECT sm."userId", sm."sourceId", 'source', NOW(), CASE WHEN sm."role" = 'blocked' THEN 'blocked' ELSE 'subscribed' END, sm."sourceId", jsonb_build_object('role', sm."role", 'referralToken', sm."referralToken")
+      FROM source_member sm
+      LIMIT ${limit} OFFSET ${offset}
```

Comment:
Do we really need batching here?

Reply:
If we do, make sure to order by time otherwise we may get corrupted data

Reply:
can't really order by time because not all entities have `createdAt`, but I will add to at least preserve order somehow

Reply:
I think we need batching because tables are quite bit and from doing it before its easier to grasp what is happening like that

Reply:
just noticed we need order by for all queries

Reply:
yes

Reply:
I can do:
- source_member.createdAt
- feed_source.sourceId
- feed_tag.tag

?

Reply:
my concern is that it's not an immutable order. it may change as we get more data.
in this case i'd suggest to snapshot the data to a temporary table and batch from there or no batching

Reply:
but why snapshot, its not breaking anything? We can always clear content preference since its not used

Reply:
@idoshamun also one thing that might not be obvious, but API is already writing to this new tables (in addition to old ones) so we just have to do it once to have data parity

Reply:
My concern is that some data will be missed due to immutable order during this migration. But if new data is being added already then it shouldn't be an issue

Reply:
yep, my bad for not highlighting that in description of the PR as well